### PR TITLE
Truncate cluster name in NTH EventBridgeRules

### PR DESCRIPTION
**Pull request by rifelpet**
ref: https://github.com/kubernetes/kops/issues/12329

alternative to https://github.com/kubernetes/kops/pull/12330

This truncates them with a random suffix and uses that shortened name in both the creation and deletion of event bridge rules. This should fix EventBridgeRule creation on clusters with long names.

### CODE REVIEW

File: tests/integration/update_cluster/nth_sqs_resources/kubernetes.tf [hunk 1]
```diff
@@ -201,193 +201,193 @@ resource "aws_autoscaling_group" "nodes-nthsqsresources-example-com" {
...
   lifecycle_transition   = "autoscaling:EC2_INSTANCE_TERMINATING"
   name                   = "nodes-NTHLifecycleHook"
 }
 
-resource "aws_cloudwatch_event_rule" "nthsqsresources-example-com-ASGLifecycle" {
-  event_pattern = file("${path.module}/data/aws_cloudwatch_event_rule_nthsqsresources.example.com-ASGLifecycle_event_pattern")
-  name          = "nthsqsresources.example.com-ASGLifecycle"
+resource "aws_cloudwatch_event_rule" "nthsqsresources-longclustername-e-fkbaoh-ASGLifecycle" {
+  event_pattern = file("${path.module}/data/aws_cloudwatch_event_rule_nthsqsresources.longclustername.e-fkbaoh-ASGLifecycle_event_pattern")
+  name          = "nthsqsresources.longclustername.e-fkbaoh-ASGLifecycle"
```
Comment (rifelpet) [inline tests/integration/update_cluster/nth_sqs_resources/kubernetes.tf:229]: here's the relevant change in the integration test

File: upup/pkg/fi/cloudup/awsup/aws_utils.go [hunk 2]
```diff
@@ -221,6 +221,15 @@ func ELBv2Tags(tags map[string]string) []*elbv2.Tag {
 	return elbv2Tags
 }
 
+// EventBridgeRulePrefix returns a suitable prefix for event bridge rules given their length restrictions
+func EventBridgeRulePrefix(cluster string) string {
```
Comment (hakman) [inline upup/pkg/fi/cloudup/awsup/aws_utils.go:225]: I would call this more like ClusterName40. But it's just a suggestion. Seems more consistent with GetResourceName32 that we use in other places.

**Review by hakman — APPROVED** [review]

**Comment by BenWolstencroft** [comment]
@rifelpet - Thanks for this fix, any idea when there will be a re-release of kops 1.21 including this fix? - we've ended up in the unenviable situation where some of our cluster estate has upgrade and some has hit this cluster hostname length issue.

**Pull request merged**